Gems/Atom/RHI/Code/Include/Atom/RHI/DrawPacket.h

```diff
@@ -44,16 +44,20 @@ namespace AZ::RHI
         //! Returns the number of draw items stored in the packet.
         size_t GetDrawItemCount() const;
 
-        //! Returns the index associated with the given DrawListTag
-        s32 GetDrawListIndex(DrawListTag drawListTag) const;
+        //! Returns the index of the first DrawItem that matches the given DrawListTag, and the given DrawFilterMask.
+        //! REMARK: When more than one MaterialPipelines are active, they may have Shaders with the same DrawListTag.
+        //!         In these cases there will be one DrawItem for each MaterialPipeline.
+        s32 GetDrawListIndex(DrawListTag drawListTag, DrawFilterMask materialPipelineMask = DrawFilterMaskDefaultValue) const;
 
         //! Returns the DeviceDrawItem at the given index
         DrawItem* GetDrawItem(size_t index);
         const DrawItem* GetDrawItem(size_t index) const;
 
-        //! Returns the DeviceDrawItem associated with the given DrawListTag
-        DrawItem* GetDrawItem(DrawListTag drawListTag);
-        const DrawItem* GetDrawItem(DrawListTag drawListTag) const;
+        //! Returns the first DrawItem that matches the given DrawListTag, and the given DrawFilterMask.
+        //! REMARK: When more than one MaterialPipelines are active, they may have Shaders with the same DrawListTag.
+        //!         In these cases there will be one DrawItem for each MaterialPipeline.
```

Comment:
@moudgils Do you know if this is expected/a supported use case? In my mind when we designed Atom a DrawPacket would never have two drawItems with the same DrawListTag, is this now a use case with MaterialPipeline?